src/BizHawk.Emulation.Cores/Waterbox/PeRunner.cs

```diff
@@ -430,13 +430,13 @@ public UIntPtr Brk(UIntPtr _p)
 			{
 				var heap = _parent._heap;
 
-				var start = heap.Memory.AddressRange.Start;
+				var start = heap.Memory.Start;
 				var end = start + heap.Used;
-				var max = heap.Memory.AddressRange.EndInclusive;
+				var max = heap.Memory.End;
 
 				var p = (ulong)_p;
 
-				if (p < start || max + 1 < p) //TODO bug?

```

Comment:
I think we went over this one in IRC a few months back.  If you read the specification of the brk syscall thoroughly, there's no bug here.  It returns a pointer to the first invalid byte, not the last valid byte.